product_supplierinfo_for_customer_sale/tests/test_product_name_search.py

```diff
@@ -0,0 +1,64 @@
+# coding: utf-8
+
+from openerp.tests.common import TransactionCase
+
+
+class TestProductNameSearch(TransactionCase):
+    """Test for:
+        - Assign a configuration customer for product.
+        - Test product name_search
+    """
+    def setUp(self):
+        super(TestProductNameSearch, self).setUp()
+        self.product = self.env['product.product'].create(
+            {'name': 'Name_product',
+             'default_code': 'code_product', })
+        self.customer = self.env.ref('base.res_partner_9')
+        self.customer_1 = self.env.ref('base.res_partner_2')
+        self.supplierinfo = self.env['product.supplierinfo']
+
+        self.supplierinfo_dict = {
+            'product_code': 'code_test',
+            'product_name': 'Name_test',
+            'name': self.customer.id,
+            'product_tmpl_id': self.product.product_tmpl_id.id,
+            'type': 'customer',
+        }
+
+    def test_10_find_product_customer_code(self):
+        """Assign a product_supplierinfo to the product and then search it
+        using name_search
+        """
+        self.assertFalse(self.product.customer_ids)
+
+        self.supplierinfo.create(self.supplierinfo_dict)
+        self.assertTrue(self.product.customer_ids)
+
+        # Search by product customer code
+        context = {'partner_id': self.customer.id,
+                   'supplierinfo_type': 'customer'}
+        product_names = self.product.with_context(
+            context).name_search(name='code_test')
+        self.assertEquals(len(product_names), 1)
+        self.assertEquals(self.product.id, product_names[0][0])
+        self.assertEquals('[code_test] Name_test', product_names[0][1])
+
+        # Search by product default code with the customer used in
+        # configuration customer
+        context = {'partner_id': self.customer.id,
+                   'supplierinfo_type': 'customer'}
+        product_names = self.product.with_context(
+            context).name_search(name='code_product')
+        self.assertEquals(len(product_names), 1)
+        self.assertEquals(self.product.id, product_names[0][0])
+        self.assertEquals('[code_test] Name_test', product_names[0][1])
+
+        # Search by product default code with a different customer used in
+        # configuration customer
+        context = {'partner_id': self.customer_1.id,
```

Comment:
Oh right!
In fact, you could use:
```python
self.product = self.env['product.product'].create(...).with_context(context)
```
to re-use it the same context for all cases but for a different context you could use:
`self.product.with_context(partner_id=self.customer_1.id`)